src/niweb/apps/noclook/forms/common.py

```diff
@@ -963,3 +980,26 @@ def __init__(self, *args, **kwargs):
         self.fields['relationship_member_of'].choices = get_node_type_tuples('Contact')
 
     relationship_member_of = relationship_field('contact', True)
+
+    def clean(self):
+        if 'relationship_member_of' in self.data:
+            self.data = self.data.copy()
+            del self.data['relationship_member_of']
+
+
+class NewRoleForm(forms.ModelForm):
+    class Meta:
+        model = Role
+        fields = '__all__'
+
+class EditRoleForm(forms.ModelForm):
+    def save(self, commit=True):
+        role = super(EditRoleForm, self).save(commit)
+        if 'name' in self.changed_data:
+            nc.models.RoleRelationship.update_roles_withid(role.handle_id, role.name)
+
```

Comment:
Do role.save() here to actual save the Role to the sql db.

Reply:
Same as above, hopefully this could be changed very soon.